packages/discovery-react-components/src/components/DocumentPreview/components/PdfViewerWithHighlight/PdfViewerWithHighlight.tsx

```diff
@@ -99,32 +101,51 @@ const PdfViewerWithHighlight = forwardRef<any, Props>(
 
     const setCurrentErrMsgFromPdfConst = useIsPfdError(_isPdfRenderError, setIsPdfRenderError);
 
+    // TODO: Dynamically create a style for every category. Match color of category
+    const styleColor = Object.values(facetInfoMap || {})
+      .map(facetInfo => {
+        return `
+        .category_${facetInfo.facetId}.mf-highlight {
+          background: ${facetInfo.color};
+          border: 2px solid ${facetInfo.color};
+        }
+        .category_${facetInfo.facetId}.mf-active {
+          background: unset;
+          opacity: 100%;
+          border: 2px solid ${facetInfo.color};
+        }`;
+      })
+      .join('\n');
+
     const highlightReady = !!documentInfo && !!renderedText;
     return (
-      <PdfViewer
-        ref={scrollRef}
-        {...rest}
-        page={currentPage}
-        setRenderedText={setRenderedText}
-        setIsPdfRenderError={setCurrentErrMsgFromPdfConst}
-      >
-        {({ fitToWidthRatio }: { fitToWidthRatio: number }) => {
-          return (
-            (state.fields || state.bboxes) && (
-              <PdfHighlight
-                parsedDocument={highlightReady ? documentInfo ?? null : null}
-                pdfRenderedText={highlightReady ? renderedText : null}
-                page={currentPage}
-                scale={scale * fitToWidthRatio}
-                highlights={state.fields}
-                boxHighlights={state.bboxes}
-                activeIds={state.activeIds}
-                {...highlightProps}
-              />
-            )
-          );
-        }}
-      </PdfViewer>
+      <>
+        <style>{styleColor}</style>
```

Comment:
Nit: rename this to `colorStyles`, since it represents a set of styles and not a color.